server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/FileEntryNode.java

```diff
@@ -26,9 +34,25 @@
 public class FileEntryNode extends Node {
 
   private final String fileName;
+  private final String fileControlClause;
 
-  public FileEntryNode(Locality location, String name) {
+  public FileEntryNode(Locality location, String name, String fileControlClause) {
     super(location, NodeType.FILE_CONTROL_ENTRY);
     this.fileName = name;
+    this.fileControlClause = fileControlClause;
+    addProcessStep(this::processNode);
   }
+
+  private List<SyntaxError> processNode() {
+    this.getNearestParentByType(ROOT)
```

Comment:
Maybe you need to find the nearest `PROGRAM` node. Otherwise, it will cause you errors in nested programs.

Reply:
Fixed